### DISCUSSION TITLE
[Feature]: Full rootless Docker support

### DISCUSSION THREAD

**Discussion by peaklabs-dev** [coollabsio/coolify#3142] [Feature Requests]
### Description

Support full rootless Docker. On localhost (coolify instance) and on connected remote servers. 
This will greatly improve the security of the coolify containers and the services running on them, as rootless Docker isolation is required to truly separate each container.
cc @DeVoresyah

### Minimal Reproduction (if possible, example repository)

-

### Version

-

**Comment by peaklabs-dev** [comment]
Good article: https://medium.com/@mccode/processes-in-containers-should-not-run-as-root-2feae3f0df3b

- This should be possible with Docker.
- A second thing to consider is implementing podman for better security, or even as an addition to docker, since it runs more securely by default (rootless). https://podman.io/

**Comment by peaklabs-dev** [comment]
Some more help to implement:
To run Docker as a router, we need to use custom Docker images for all services (so we essentially need to rewrite the services a bit) as we need to use the yml file and the a custom dockerfile for each service:
Issue that adds this: https://github.com/coollabsio/coolify/issues/2552 
1. In each service folder there should be a service.yml file and a custom service.dockerfile, so for example for WordPress: wordpress.yml and DOCKERFILE inside the wordpress service folder.
2. In the custom docker file we use the flowing command to create a random user that does not have root privileges and disable the root user:
```Dockerfile
RUN groupadd -r <user> && useradd -r -g <group> <user>
RUN chsh -s /usr/sbin/nologin root
```
3. In the yml file we access the image something like this:
```yml
service:
    build:
      context: .
```
@andrasbacsai if you need more help just let me know.

**Comment by roo12312** [comment]
+1
